third_party/amd/lib/TritonAMDGPUToLLVM/MemoryOpToLLVM.cpp

```diff
@@ -102,10 +102,129 @@ struct LocalLoadOpConversion
+      }
+    }
+    return failure();
+  }
+
+private:
+  bool canUseTransLoad(MemDescType srcTy, RankedTensorType dstTy) const {
+    auto bitwidth = typeConverter->convertType(dstTy.getElementType())
+                        .getIntOrFloatBitWidth();
+
+    auto dotEnc = llvm::cast<DotOperandEncodingAttr>(dstTy.getEncoding());
+    auto mfmaEnc = llvm::dyn_cast<AMDMfmaEncodingAttr>(dotEnc.getParent());
+    // Currently, only transpose loading for 16-bit element types is supported.
+    // Support for 8-bit types will be added in a future patch.
+    if (!mfmaEnc || bitwidth != 16) {
+      return false;
+    }
+
+    // The transposed load lowering logic assumes that double-rate MFMA
+    // instructions are used whenever possible. This code verifies whether
+    // double-rate MFMA instructions are being used and falls back to the
+    // default path if they are not. (Note: The lowering logic for double-rate
+    // MFMA is the same as for single-rate with kpack=2). This check should be
+    // removed once double-rate MFMA support is fully implemented in the
+    // compiler, leaving only an assertion. Currently, single-rate
+    // configurations with kpack=1 are still in use, so in such cases, we revert
+    // to the default lowering logic without LDS transposed read instructions.
+    int rank = dstTy.getRank();
+    int32_t kWidth = dotEnc.getKWidth();
+    const int32_t mDim = mfmaEnc.getMDim();
+    assert((mDim == 32 || mDim == 16) && "Invalid MFMA instruction dimension");
+
+    const int largeTileThreshold = (mDim == 32) ? 8 : 16;
+    const auto shape = dstTy.getShape();
+    const int kDim = dotEnc.getOpIdx() == 0 ? rank - 1 : rank - 2;
+
+    const bool isLargeTile = shape[kDim] > largeTileThreshold;
+    const int expectedKWidth = isLargeTile ? 8 : 4;
+
+    if (kWidth != expectedKWidth)
+      return false;
+
+    auto sharedEnc =
+        dyn_cast<triton::gpu::SwizzledSharedEncodingAttr>(srcTy.getEncoding());
+    if (!sharedEnc)
+      return false;
+
+    bool nonKContig = kDim != sharedEnc.getOrder()[0];
+    return nonKContig && targetInfo.canUseLDSTransLoad(bitwidth);
+  }
+
+  LogicalResult
+  lowerSharedToDotOperandTransLL(triton::gpu::LocalLoadOp op,
+                                 triton::gpu::LocalLoadOpAdaptor adaptor,
+                                 const LLVMTypeConverter *typeConverter,
+                                 ConversionPatternRewriter &rewriter) const {
+    auto ctx = rewriter.getContext();
+    auto loc = op.getLoc();
+    auto b = TritonLLVMOpBuilder(loc, rewriter);
+    auto dstTy = cast<RankedTensorType>(op.getType());
+    auto srcTy = cast<MemDescType>(op.getSrc().getType());
+    auto dotEnc = cast<DotOperandEncodingAttr>(dstTy.getEncoding());
+    auto shape = dstTy.getShape();
+
+    auto llvmElemTy = typeConverter->convertType(dstTy.getElementType());
+    auto bitwidth = llvmElemTy.getIntOrFloatBitWidth();
+    auto ldsTransLayout = chooseLDSTransLayout(dotEnc, shape, bitwidth);
```

Comment:
The names (both the variable and function names) are misleading. It sounds like a new layout for LDS. Actually, it's a regLayout, as indicated by the API of `emitTransferBetweenRegistersAndShared`. 
We can follow nv naming converntion, i.e. include the ds_read_b64_tr_b16 instruction name in the function name, such as `chooseDsReadB64Tr16Layout`. Then for the variable, we can name it as `dsReadTransLayout`.

I'd suggest we name the function as above, i.e. expose the instruction details. It's aligned with the current limitation that we only support 16-bit trans load. Later, when we support more bitWidth, we can see if it's easy to extend the function in place, or we may need add more `chooseXXX` functions for each trans load bitWidth.